lib/matplotlib/tests/test_collections.py

```diff
@@ -830,6 +831,37 @@ def test_collection_set_verts_array():
         assert np.array_equal(ap._codes, atp._codes)
 
 
+@pytest.mark.parametrize("fname", ["fill_between", "fill_betweenx"])
+@check_figures_equal(extensions=["png"])
+def test_fbp_collection_set_data(fname, fig_test, fig_ref):
+    t = np.linspace(0, 16)
+    f1 = np.sin(t)
+    f2 = f1 + 0.2
+    f3 = f2.copy()
+    f3[10], f3[20] = f3[20], f3[10]
+
+    ax_ref = fig_ref.subplots()
+    getattr(ax_ref, fname)(t, f1, f2)
+
+    ax2 = fig_test.subplots()
+    coll2 = getattr(ax2, fname)(t, f1, f3)
+    coll2.set_data(t, f1, f2)
```

Comment:
We can make this even simpler:
- don't need `f3`. Creating arbitrary input on the fly is good enough
- This tests `set_data`. It is conceptually not entangled with `t_direction`, which only affects rendering. We thus don't have to test with both `fill_between` and `fill_betweenx`.

```suggestion
@check_figures_equal(extensions=["png"])
def test_fbp_collection_set_data(fig_test, fig_ref):
    t = np.linspace(0, 16)
    f1 = np.sin(t)
    f2 = f1 + 0.2

    fig_ref.subplots().fill_between(t, f1, f2)
    coll = fig_test.subplots().fill_between(t, t, 2*t)
    coll2.set_data(t, f1, f2)
```

Reply:
- Partially adapted in 13cad5d88bf8ee540be69465b1cde48dfbd6f3a3
- `coll = fig_test.subplots().fill_between(t, t, 2*t)` leads to a failed comparison, because the plot ranges are different.

Reply:
You could do: 
```
    fig_ref.subplots().fill_between(t, f1, f2)
    # preliminary constant data with the same scale as f1, f2
    # to ensure y-axis scaling is consistent
    coll = fig_test.subplots().fill_between(t, -1, 1.2)
    coll2.set_data(t, f1, f2)
```

Reply:
13cad5d88bf8ee540be69465b1cde48dfbd6f3a3